Add SAP HANA schema collection and diagnostics (#23934)
DataDog/integrations-core: sap_hana/README.md

   GRANT SELECT ON SYS_DATABASES.M_VOLUME_IO_TOTAL_STATISTICS TO DD_MONITOR;
   ```

>    To collect schema metadata for Database Monitoring (see [Schema collection](#schema-collection)), also grant select on the catalog views that store schema, table, and column definitions. These are already covered by the `GRANT CATALOG READ` in step 2, so this is only needed if you skipped that grant:

Reviewer: ```suggestion
   To collect schema metadata for Database Monitoring (see [Schema collection](#schema-collection)), grant select on the catalog views that store schema, table, and column definitions. These are already covered by the `GRANT CATALOG READ` in step 2, so this is only needed if you skipped that grant:
```

Author: Thanks! The grant note is in place. We settled on "Data Quality features in Data Observability" wording in a later revision rather than "Database Monitoring", but kept your structure of pointing back to the Granting privileges section and noting it is covered by \`GRANT CATALOG READ\`.

3. [Restart the Agent][5].

#### Schema collection

> The Agent can collect SAP HANA catalog metadata (schemas, tables, and columns) for Database Monitoring's Schema Explorer. Collection is disabled by default. To enable it, make sure the monitoring user can read the catalog views (see [Granting privileges](#granting-privileges)) and add the following block to your `sap_hana.d/conf.yaml` file:

Reviewer: ```suggestion
The Agent can collect SAP HANA catalog metadata (schemas, tables, and columns) for Database Monitoring's Schema Explorer. Collection is disabled by default. To enable it, ensure that the monitoring user can read the catalog views (see [Granting privileges](#granting-privileges)) and add the following block to your `sap_hana.d/conf.yaml` file:
```

Author: Same as above — adopted your guidance to point users at the Granting privileges section and note collection is disabled by default, with the wording aligned to "Data Quality features in Data Observability".

3. [Restart the Agent][5].

#### Schema collection

> The Agent can collect SAP HANA catalog metadata (schemas, tables, views, and columns) for Data Quality features in Data Observability. When the monitoring user has access to `SYS.M_TABLE_STATISTICS`, the Agent also collects row counts and last modification times for tables. Collection is disabled by default. To enable it, ensure that the monitoring user can read the required views (see [Granting privileges](#granting-privileges)) and add the following block to your `sap_hana.d/conf.yaml` file:

Reviewer 2: ```suggestion
The Agent can collect SAP HANA catalog metadata (schemas, tables, views, and columns) for Data Quality features in Data Observability. When the monitoring user has access to `SYS.M_TABLE_STATISTICS`, the Agent also collects row counts and last modification times for tables. Collection is disabled by default. To enable schema collection, ensure that the monitoring user can read the required views (see [Granting privileges](#granting-privileges)) and add the following block to your `sap_hana.d/conf.yaml` file:
```
